spring-pulsar/src/test/java/org/springframework/pulsar/core/DefaultTopicResolverTests.java

```diff
@@ -110,7 +110,7 @@ static Stream<Arguments> resolveByMessageTypeProvider() {
 				arguments("complexMessageWithUserTopic", userTopic, Foo.class, defaultTopic, userTopic),
 				arguments("complexMessageNoUserTopic", null, Foo.class, defaultTopic, fooTopic),
 				arguments("nullMessageWithUserTopicAndDefault", userTopic, null, defaultTopic, userTopic),
-				arguments("nullMessageWithDefault", null, null, defaultTopic, defaultTopic),
+				arguments("nullMessageWithDefault", null, null, defaultTopic, null),
```

Comment:
Is it OK ?
Or do we prefer to return the default topic when the message is null ?

Reply:
I'm not sure. Up until now we have returned default when message is null. 

1) If we return the default topic then tombstones will go to the default topic (which is probably not the best choice). 
2) If we return empty then they get a clear answer that they need to specify the topic.

Of these I would pick 2) as bad things can happen if we start shipping nulls to a topic a user configured as the "default" (but did not mean for the null records).

Where else can the message be null that calls in from here? What about the consumer containers? Will this affect them in any way? They have message type though IIRC.

Reply:
On the consumer path, we get the type by reflection so it won't be null.